From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Mon, 24 Apr 2023 15:37 +0100

Change the size of the Linux poolsize from 4096 to 256 to reflect
that of modern/recent Linux kernels.

Upstream PR #196 submitted.

---

diff -aur a/rngd.8.in b/rngd.8.in
--- a/rngd.8.in
+++ b/rngd.8.in
@@ -108,7 +108,7 @@
 Setting this too high will cause \fIrngd\fR to dominate the contents of the
 entropy pool.  Low values will hurt system performance during entropy
 starves.  Do not set \fIfill-watermark\fR above the size of the
-entropy pool (usually 4096 bits).  A value of 0 to this option will cause no
+entropy pool (usually 256 bits).  A value of 0 to this option will cause no
 watermark to be set.
 .TP
 \fB\-R\fI n\fR, \fB\-\-force-reseed=\fInnn\fR
diff -aur a/rngd.c b/rngd.c
--- a/rngd.c
+++ b/rngd.c
@@ -120,7 +120,7 @@
 	  "Number of bytes written to random-device at a time (default: 64)" },
 
 	{ "fill-watermark", 'W', "n", 0,
-	  "Do not stop feeding entropy to random-device until at least n bits of entropy are available in the pool (default: 3/4 of poolsize), 0 <= n <= 4096" },
+	  "Do not stop feeding entropy to random-device until at least n bits of entropy are available in the pool (default: 3/4 of poolsize), 0 <= n <= 256" },
 
 	{ "quiet", 'q', 0, 0, "Suppress all messages" },
 
@@ -628,7 +628,7 @@
 		break;
 	case 'W': {
 		int n;
-		if ((sscanf(arg, "%i", &n) == 0) || (n < 0) || (n > 4096))
+		if ((sscanf(arg, "%i", &n) == 0) || (n < 0) || (n > 256))
 			argp_usage(state);
 		else
 			arguments->fill_watermark = n;
diff -aur a/rngd_linux.c b/rngd_linux.c
--- a/rngd_linux.c
+++ b/rngd_linux.c
@@ -60,7 +60,7 @@
  * randomdev is the path to the random device
  */
 
-#define DEFAULT_WATERMARK_GUESS 4096
+#define DEFAULT_WATERMARK_GUESS 256
 
 void init_kernel_rng(const char* randomdev)
 {
